.github/workflows/hrm-lambda-release.yml

```diff
@@ -0,0 +1,113 @@
+        default: development
+        required: true
+        type: choice
+        options:
+          - development
+          - staging
+          - production
+
+      send-slack-message:
+        description: 'Specifies if should send a Slack message at the end of successful run. Defaults to true'
+        required: true
+        type: boolean
+        default: true
+
+  workflow_call:
+    secrets:
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+      # Should probably use the passed in region for everything but the GITHUB_ACTIONS_SLACK_BOT_TOKEN SSM parameter only exists in us-east-1
+      AWS_DEFAULT_REGION:
+        required: true
+
+env:
+  GITHUB_SHA: ${{ github.sha }}
+
+jobs:
+  build-lambda:
+    name: Build And Deploy Lambda
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14.x
+
+      #TODO: Move to a matrix based region split for the ECR/update lambda steps
+      - name: Set Env Based on Environment
+        # this is overcomplicated to allow for testing with a push event
+        run: |
+          INPUT_ENVIRONMENT=${{ inputs.environment }}
+          INPUT_LAMBDA_NAME=${{ inputs.lambda_name }}
+          ENVIRONMENT=${INPUT_ENVIRONMENT:-development}
+          LAMBDA_NAME=${INPUT_LAMBDA_NAME:-retrieve-transcript}
+
+          echo ENVIRONMENT=$ENVIRONMENT >> $GITHUB_ENV
+          echo LAMBDA_NAME=$LAMBDA_NAME >> $GITHUB_ENV
+
+          case "$ENVIRONMENT" in development) region="us-east-1" ;; staging) region="eu-west-1" ;; production) region="eu-west-1" ;; *) echo "Invalid environment: ${{ inputs.environment }}" && exit 1 ;; esac
+          echo AWS_REGION=$region >> $GITHUB_ENV
+          echo AWS_DEFAULT_REGION=$region >> $GITHUB_ENV
+          echo DOCKER_IMAGE_BASE=712893914485.dkr.ecr.${region}.amazonaws.com/${ENVIRONMENT}/hrm-lambdas- >> $GITHUB_ENV
+
+      - name: Compile Lambda
+        run: |
+          cd lambdas
+          npm ci
+          npm run compile:${{ env.LAMBDA_NAME }}
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      # build and push the image
+      - name: Build and Push Docker Image
+        uses: docker/build-push-action@v2
+        with:
+          context: ./lambdas/src/${{ env.LAMBDA_NAME }}
+          file: ./lambdas/Dockerfile
+          push: true
+          tags: ${{ env.DOCKER_IMAGE_BASE }}${{ env.LAMBDA_NAME }}:live
```

Comment:
This might be a dumb question, but what the `:live` suffix stands for? Is that something we may want to specify as input?

Reply:
In the current setup where every environment will have an ECR repo in every region, it is just a hardcoded tag for "this is the docker image tag that is currently running in this enviornment". I'm not sure I can see a use case for changing it on the fly.

In the setup I'd suggest moving to at some point, where there is one public repo for each lambda type, the substitution will change. It will be something like: `public.ecr.aws/techmatters/hrm-lambdas-${{ env.LAMBDA_NAME }}:${{ env.ENVIRONMENT }}` so you can look at docker tags and see what image is running in each environment.